tests/test_scheduler_worker.py

```diff
@@ -57,19 +57,30 @@ def test_mat_add(scheduler_client, matrix1, matrix2):
 
 def test_mat_subtract(scheduler_client, matrix1, matrix2):
     task = Task(taskId="2", taskDefintion="mat_subtract", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()])
-    future = scheduler_client.SubmitTask(task)
+    future = scheduler_client.SubmitTask(task)[0]
 
     worker_client = WorkerClient(future.hostName, future.port)
 
     expected_result = [[-4, -4 ], [-4, -4]]
     poll_for_result(worker_client, future, expected_result)
 
 def test_passing_futures_as_args_flow(scheduler_client, matrix1, matrix2):
-    future = scheduler_client.SubmitTask(Task(taskId="0", taskDefintion="dot_product", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))
-    future2 = scheduler_client.SubmitTask(Task(taskId="1", taskDefintion="mat_add", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))
-    future3 = scheduler_client.SubmitTask(Task(taskId="2", taskDefintion="mat_subtract", taskData=[future, future2]))
+    future = scheduler_client.SubmitTask(Task(taskId="0", taskDefintion="dot_product", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))[0]
+    future2 = scheduler_client.SubmitTask(Task(taskId="1", taskDefintion="mat_add", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))[0]
+    future3 = scheduler_client.SubmitTask(Task(taskId="2", taskDefintion="mat_subtract", taskData=[future, future2]))[0]
 
     workerClient = WorkerClient(future3.hostName, future3.port)
 
     expected_result = [[13, 14], [33, 38]]
     poll_for_result(workerClient, future3, expected_result)
+
+# Multiple workers here doensn't guarantee that the task will be assigned to different workers. It depends on the scheduling mode and the number of workers registered.
+# Make sure --AssingedWorkersPerTask is set to more than 1 to ideally test this(not assigning would still work).
+def test_assign_task_to_multiple_workers(scheduler_client, matrix1, matrix2):
+    futures = scheduler_client.SubmitTask(Task(taskId="0", taskDefintion="dot_product", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))
+    expected_result = [[19, 22], [43, 50]]
+
+    for future in futures:
+        logging.info(f"Future: {future}")
+        worker_client = WorkerClient(future.hostName, future.port)
+        poll_for_result(worker_client, future, expected_result)
```

Comment:
also add a test where submit/get_result times out to make sure it is working as expected and we dont break it in future?

Reply:
We can't guarantee that with the current --addDelay arg in Worker since it randomises delay which can't guarantee timeouts. 

Idea: Introduce --addCustomDelay to have a custom delay set and this will lead to timing out every time.
Do we really need this tho?

Reply:
Lets take deterministic addDelay as a follow up item